community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/NodeCorrectlyIndexedCheck.java

```diff
@@ -114,6 +117,24 @@ private void verifyNodeCorrectlyIndexedUniquely(
         }
     }
 
+    private PrimitiveLongIterator filterIfMultipleValuesFound( int propertyKeyId, Object propertyValue,
+            PrimitiveLongIterator indexedNodeIds )
+    {
+        if ( ((PrimitiveLongPeekingIterator) indexedNodeIds).hasMultipleValues() )
```

Comment:
This is used to speed up the CC. As CC is offline, we could assume that if we find only one match in the first stage, then the value is the same, a.k.a. the node is indexed uniquely. The trick here is that this CC filtering is called in a `verify` method. For the `search` method, we must return a exact match, and that is what we do in the "normal" filtering.